flexmeasures/data/models/planning/__init__.py

```diff
@@ -1,11 +1,144 @@
+              E.g. start and flex_model[soc_at_start] are intertwined.
+        """
+        self.sensor = sensor
+        self.start = start
+        self.end = end
+        self.resolution = resolution
+        self.belief_time = belief_time
+        self.round_to_decimals = round_to_decimals
+        if flex_model is None:
+            flex_model = {}
+        self.flex_model = flex_model
+        if flex_context is None:
+            flex_context = {}
+        self.flex_context = flex_context
+
+    def compute_schedule(self) -> Optional[pd.Series]:
+        """
+        Overwrite with the actual computation of your schedule.
+        """
         return None
+
+    @classmethod
+    def get_data_source_info(cls: type) -> dict:
+        """
+        Create and return the data source info, from which a data source lookup/creation is possible.
+        See for instance get_data_source_for_job().
+        """
+        source_info = dict(
+            model=cls.__name__, version="1", name="Unknown author"
+        )  # default
+
+        if hasattr(cls, "__version__"):
+            source_info["version"] = str(cls.__version__)
+        else:
+            current_app.logger.warning(
+                f"Scheduler {cls.__name__} loaded, but has no __version__ attribute."
+            )
+        if hasattr(cls, "__author__"):
+            source_info["name"] = str(cls.__author__)
+        else:
+            current_app.logger.warning(
+                f"Scheduler {cls.__name__} has no __author__ attribute."
+            )
+        return source_info
+
+    def persist_flex_model(self):
+        """
+        If useful, (parts of) the flex model can be persisted (e.g on the sensor) here,
+        e.g. as asset attributes, sensor attributes or as sensor data (beliefs).
+        """
+        pass
+
+    def inspect_config(self):
+        """
+        Check all configurations we have, throwing either ValidationErrors or ValueErrors.
+        Other code can decide if/how to handle those.
+        """
+        self.inspect_timing_config()
+        self.inspect_flex_config()
+        self.config_inspected = True
+
+    def inspect_timing_config(self):
+        """
+        Check if the timing of the schedule is valid.
+        Raises ValueErrors.
+        """
+        if self.start > self.end:
+            raise ValueError(f"Start {self.start} cannot be after end {self.end}.")
+        # TODO: check if resolution times X fits into schedule length
+        # TODO: check if scheduled events would start "on the clock" w.r.t. resolution (see GH#10)
+
+    def inspect_flex_config(self):
+        """
+        Check if the flex model and context are valid. Should be overwritten.
+
+        Ideas:
+        - Apply a schema to check validity (see in-built flex model schemas)
+        - Check for inconsistencies between settings (can also happen in Marshmallow)
+        - fill in missing values from the scheduler's knowledge (e.g. sensor attributes)
```

Comment:
FYI: I found out how to access the sensor object (deserialized from `location="path"`) in the schema used for deserializing fields from `location="json"` (using https://webargs.readthedocs.io/en/latest/advanced.html#meta-locations). Knowing this, filling in missing values can also happen using Marshmallow, e.g. using `@validates(field_name)` or `@validates_schema`. So all three of these ideas can happen in Marshmallow, which may be worth a note and/or a follow-up ticket.

Btw, I started using these meta-locations in https://github.com/FlexMeasures/flexmeasures/pull/563, so I know how to make that work now.